crates/trie/db/src/trie_cursor.rs

```diff
@@ -158,7 +158,7 @@ where
             if let Some(node) = maybe_updated {
                 self.cursor.upsert(
                     self.hashed_address,
-                    StorageTrieEntry { nibbles, node: node.clone() },
+                    &StorageTrieEntry { nibbles, node: node.clone() },
```

Comment:
could this potentially be `StorageTrieEntryRef` so that `node` is not cloned either?

Reply:
yeah this will be unlocked by this